synapse/handlers/auth.py

```diff
@@ -1950,7 +1946,11 @@ def run(*args: Tuple, **kwargs: Dict) -> Awaitable:
         # need to use a tuple here for ("password",) not a list since lists aren't hashable
         auth_checkers[(LoginType.PASSWORD, ("password",))] = check_password
 
-    api.register_password_auth_provider_callbacks(hooks, auth_checkers=auth_checkers)
+    api.register_password_auth_provider_callbacks(
+        auth_checkers=auth_checkers,
+        check_3pid_auth=check_3pid_auth_hook,
+        on_logged_out=on_logged_out_hook,
```

Comment:
Let's keep them the same order as the definition:

```suggestion
        check_3pid_auth=check_3pid_auth_hook,
        on_logged_out=on_logged_out_hook,
        auth_checkers=auth_checkers,
```